### PR TITLE
Create bitcoin_utils.py to make bitcoin payments available to agents 

### PR DESCRIPTION
**Pull request by dbscooper**
### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thouroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — APPROVED** [review]
Even if it's useless it's a good self-contained module to use later on.

**Comment by benthecarman** [comment]
Shouldnt this also need a get address function?

File: bitcoin_utils.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    recipient_address = P2PKHBitcoinAddress(recipient_address)
+
+    amount_satoshis = int(amount_btc * 100000000)
+    fee_satoshis = int(fee_btc * 100000000)
+
+    unspent = [utxo for utxo in rpc.listunspent(0) if utxo['address'] == sender_address]
+    if not unspent:
+        raise Exception('No available UTXOs for the sender address.')
+
+    unspent.sort(key=lambda x: -x['amount'])
```
Comment (benthecarman) [inline bitcoin_utils.py:67]: you should randomize the list, if you sort by amount you'll either grind your utxos to dust or pay way too much fees

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
It's not a good idea to keep adding things to a reviewed commit without a reviewer's request. Create another commit instead and say that it's dependent on this one. @Torantulino, @richbeales , @p-i- ?

File: bitcoin_utils.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    address = P2PKHBitcoinAddress.from_pubkey(secret_key.pub)
+    return str(address)
+
+
+def send_bitcoin(sender_secret, recipient_address, amount_btc, fee_btc, context):
+    network = choose_network(context)
+    SelectParams(network)
+    rpc = Proxy()
+
+    # Rest of the send_bitcoin function implementation
```
Comment (benthecarman) [inline bitcoin_utils.py]: think you forgot to do this

**Review by nponeccop — APPROVED** [review]

**Comment by talvasconcelos** [comment]
Just my two sats on this, but i thing Bitcoin's Lightning Network would more appropriate for this.

**Comment by Pwuts** [comment]
This is out of scope for the core project (since #757) but could be a plugin!

**Pull request closed without merging**